bokeh/models/markers.py

```diff
@@ -240,161 +235,135 @@ class Circle(Marker):
 
-    __example__ = "examples/reference/models/SquarePin.py"
-
-class SquareCross(Marker):
+def SquareCross(*args, **kwargs):
     ''' Render square markers with a '+' cross through the center. '''
+    deprecated((2, 3, 0), "SquareCross()", "Scatter(marker='square_cross')")
+    return Scatter(*args, **kwargs, marker="square_cross")
 
-    __example__ = "examples/reference/models/SquareCross.py"
-
-class SquareX(Marker):
+def SquareX(*args, **kwargs):
     ''' Render square markers with an 'X' cross through the center. '''
+    deprecated((2, 3, 0), "SquareX()", "Scatter(marker='square_x')")
+    return Scatter(*args, **kwargs, marker="square_x")
 
-    __example__ = "examples/reference/models/SquareX.py"
-
-class Triangle(Marker):
+def Triangle(*args, **kwargs):
     ''' Render triangle markers. '''
+    deprecated((2, 3, 0), "Triangle()", "Scatter(marker='triangle')")
+    return Scatter(*args, **kwargs, marker="triangle")
 
-    __example__ = "examples/reference/models/Triangle.py"
-
-class TriangleDot(Marker):
+def TriangleDot(*args, **kwargs):
     ''' Render triangle markers with center dots. '''
+    deprecated((2, 3, 0), "TriangleDot()", "Scatter(marker='triangle_dot')")
+    return Scatter(*args, **kwargs, marker="triangle_dot")
 
-    __example__ = "examples/reference/models/TriangleDot.py"
-
-class TrianglePin(Marker):
+def TrianglePin(*args, **kwargs):
     ''' Render pin-cushion triangle markers. '''
+    deprecated((2, 3, 0), "TrianglePin()", "Scatter(marker='triangle_pin')")
+    return Scatter(*args, **kwargs, marker="triangle_pin")
 
-    __example__ = "examples/reference/models/TrianglePin.py"
-
-class X(Marker):
+def X(*args, **kwargs):
     ''' Render 'X' markers. '''
+    deprecated((2, 3, 0), "X()", "Scatter(marker='x')")
+    return Scatter(*args, **kwargs, marker="x")
 
-    __example__ = "examples/reference/models/X.py"
-
-class Y(Marker):
+def Y(*args, **kwargs):
     ''' Render 'Y' markers. '''
-
-    __example__ = "examples/reference/models/Y.py"
+    deprecated((2, 3, 0), "Y()", "Scatter(marker='y')")
+    return Scatter(*args, **kwargs, marker="y")
 
 #-----------------------------------------------------------------------------
 # Dev API
 #-----------------------------------------------------------------------------
 
-marker_types = {
-    "asterisk": Asterisk,
-    "circle": Circle,
-    "circle_cross": CircleCross,
-    "circle_dot": CircleDot,
-    "circle_x": CircleX,
-    "circle_y": CircleY,
-    "cross": Cross,
-    "dash": Dash,
-    "diamond": Diamond,
-    "diamond_cross": DiamondCross,
-    "diamond_dot": DiamondDot,
-    "dot": Dot,
-    "hex": Hex,
-    "hex_dot": HexDot,
-    "inverted_triangle": InvertedTriangle,
-    "plus": Plus,
-    "square": Square,
-    "square_cross": SquareCross,
-    "square_dot": SquareDot,
-    "square_pin": SquarePin,
-    "square_x": SquareX,
-    "triangle": Triangle,
-    "triangle_dot": TriangleDot,
-    "triangle_pin": TrianglePin,
-    "x": X,
-    "y": Y,
-}
-
```

Comment:
I think `marker_type` should be kept for now (at least as a list of the names) and then removed at 3.0